src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/ConfiguredExceptionsStoreTests.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Diagnostics.Monitoring.TestCommon;
+using Microsoft.Diagnostics.Monitoring.WebApi.Exceptions;
+using Microsoft.Diagnostics.Tools.Monitor.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    public sealed class ConfiguredExceptionsStoreTests
```

Comment:
Could we add a test with a `TopLevelLimit` of 2 (or just not 1)? If we want to keep it simple we could probably just add three outer exceptions and make sure the initial one is dropped